k8s/operators/src/diskpool/main.rs

```diff
@@ -600,13 +590,32 @@ impl ResourceContext {
             |event| async move {
                 match event {
                     finalizer::Event::Apply(dsp) => Self::put_finalizer(dsp).await,
-                    finalizer::Event::Cleanup(_dsp) => Self::delete_finalizer(self.clone()).await,
+                    finalizer::Event::Cleanup(dsp) => {
+                        match self
+                            .pools_api()
+                            .get_node_pool(&self.spec.node(), &self.name_any())
+                            .await
+                        {
+                            Ok(pool) => {
+                                if dsp.status.as_ref().unwrap().cr_state == CrPoolState::Created {
+                                    let new_status = DiskPoolStatus::terminating(pool.into_body());
+                                    let _ = self.patch_status(new_status).await?;
+                                }
```

Comment:
We want to show Pool usage when marking Terminating. Since we couldnt get pool from the control plane we just delete finalizer for that cr and remove from inventory. We dont need to transition to Terminating for this IMO. There should be nothing holding off on directly deleting cr?

Reply:
What if the pool has replicas? Or if the node is offline? Then we cannot delete the pool, so meanwhile we should show the `cr_state` as `Terminating`, even if the if the previous state was `Creating` and not `Created`? Otherwise it will stay as `Creating` even after user issued delete, which doesn't seem right to me.

@blaisedias is this covered by the BDD, and if so what is the expected behaviour?

Reply:
@tiagolobocastro We will tarnsition to Terminating now on all states. Made changes

Reply:
It is indeed covered by the BDD- create and a delete on a k8s node not provisioned with the IO engine.